### PR TITLE
[PAY-1589] Wire up Stripe Onramp in mobile

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Wire up stripe onramp flow in mobile using a webview. Adds a libs fn to create a stripe session via identity. Still using sol bc haven't updated the identity endpoint to support usdc yet.

### Dragons

- cc @dylanjeffers to check package-lock changes, followed your steps from last time so hopefully it's good!
- Duplicating code in `buyCrypto.ts` that already exists in web in `BuyAudio.ts`, but using different libs so don't think this makes sense to share.

### How Has This Been Tested?

Local ios stage

https://github.com/AudiusProject/audius-client/assets/3893871/cee9ae67-e0aa-4eb6-bbba-99fc26159856

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        amount: price,
+        destinationWallet: 'FgMM3AG1qfpEEdYUiw4Zk9ouqsTkheC3rsu4RjvKtyEB'
+      })
+      if (res !== undefined && res.client_secret !== undefined) {
+        navigation.navigate('StripeOnrampEmbed', {
+          clientSecret: res.client_secret
+        })
+      }
+    } catch (e) {
+      console.log(e)
```
Comment (rickyrombo) [inline packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:34]: console.error?

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  price
+}: StripePurchaseConfirmationButtonProps) => {
+  const navigation = useNavigation()
+  const { specialLightGreen1 } = useThemeColors()
+
+  const handleBuyPress = useCallback(async () => {
+    try {
+      const res = await createStripeSession({
+        amount: price,
+        destinationWallet: 'FgMM3AG1qfpEEdYUiw4Zk9ouqsTkheC3rsu4RjvKtyEB'
```
Comment (rickyrombo) [inline packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:26]: See if we can't derive this using `solanaWeb3Manager` - latest libs should have it updated

File: packages/mobile/.env.stage [hunk 3]
```diff
@@ -76,3 +76,5 @@ FINGERPRINT_PUBLIC_API_KEY=Rz2A3Y5YGSg9K80VgKPi
 FINGERPRINT_ENDPOINT=https://fp.staging.audius.co
 
 OLD_WEB_APP_STATIC_SERVER_PORT=3101
+
+REACT_APP_STRIPE_CLIENT_PUBLISHABLE_KEY=pk_test_51LPsGuCJOWtpH6AEZT3Wf2U2xmLZQrEV56yha7HEVTEyhYYVrWCdknml3t4gkSe9Nagd1o9Royy8zL3XEAmRzeHS00xAKTfgpi
```
Comment (rickyrombo) [inline packages/mobile/.env.stage:80]: Also add to .env.prod?

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx (hunk 2 above)
Comment (dharit-tan) [reply packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:26]: oops my b lol

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+export const StripePurchaseConfirmationButton = ({
+  price
+}: StripePurchaseConfirmationButtonProps) => {
+  const navigation = useNavigation()
+  const { specialLightGreen1 } = useThemeColors()
+
+  const handleBuyPress = useCallback(async () => {
+    try {
+      const res = await createStripeSession({
```
Comment (schottra) [inline packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:24]: So, isn't there already a function in web that does this? I was hoping we could move things like that to common.

File: packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  const { clientSecret } = params
+  const isUSDCEnabled = useIsUSDCEnabled()
+
+  const handleSessionUpdate = useCallback((event) => {
+    if (event?.payload?.session?.status) {
+      console.log(`Stripe Session Update ${event.payload.session.status}`)
+    }
+  }, [])
+
+  const html = `
```
Comment (schottra) [inline packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: I don't suppose there's a way we could put this into an actual html file and import it? So we get a better editing experience.

**Review by schottra — APPROVED** [review]
Very cool!

File: packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx (hunk 5 above)
Comment (dharit-tan) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: spoke with @rickyrombo I think we'd have to host the file somewhere. Totally agree but just think it's not worth the hassle.

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx (hunk 4 above)
Comment (dharit-tan) [reply packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:24]: Agree. There is an equivalent fn in web, but libs there is attached to window whereas in mobile it's not. We _could_ pass in libs I guess but that feels a bit weird/not sure if that's a pattern we want to follow. Any thoughts @rickyrombo ?

File: packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx (hunk 5 above)
Comment (rickyrombo) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: I think this is also doable actually!
https://stackoverflow.com/questions/42851296/react-native-load-local-html-file-into-webview

File: packages/mobile/src/services/buyCrypto.ts [hunk 6]
```diff
@@ -1,5 +1,9 @@
 import { audiusLibs, waitForLibsInit } from './libs'
 
+export const getUSDCUserBank = async (ethWallet: string) => {
+  return await audiusLibs?.solanaWeb3Manager?.deriveUserBank(ethWallet, 'usdc')
```
Comment (rickyrombo) [inline packages/mobile/src/services/buyCrypto.ts:4]: nit: `waitForLibsInit()`?

Comment (dharit-tan) [reply packages/mobile/src/services/buyCrypto.ts:5]: great call

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx (hunk 4 above)
Comment (rickyrombo) [reply packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:39]: need to think on this one more. I think what we have here is good for now, but we definitely need to establish a "meta" for libs/sdk

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx (hunk 5 above)
Comment (dharit-tan) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: omg!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-stripe-mobile

File: packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx (hunk 5 above)
Comment (dharit-tan) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: seeing ppl in the comments complaining that this doesn't work on android. just tried testing on android but build failed :(. would like to merge this for now and create a ticket to follow-on + test on android.

Comment (dharit-tan) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: https://linear.app/audius/issue/PAY-1663/rn-webview-load-stripe-html-from-separate-file

Comment (rickyrombo) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: yeah low pri, I don't think we're going to be iterating on this much.

**Review by rickyrombo — APPROVED** [review]
Good stuff!

File: packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx (hunk 5 above)
Comment (schottra) [reply packages/mobile/src/components/stripe-onramp-embed/StripeOnrampEmbed.tsx:30]: FWIW, I wasn't imaging bundling the file and loading it from an external location at runtime. I had thought we could just use our build tooling to embed the HTML into the module at build time. It was more a developer experience suggestion than anything as having a separate HTML file makes it easier to maintain. But this is fine for now!

**Pull request merged**